### ISSUE TITLE
bug of background activation of high accuracy mode

### ISSUE THREAD

**Issue by DOK87** [home-assistant/android#4749]
Apparently activating the companion app's high accuracy mode works the opposite of how it should.

I configured the companion app to activate high accuracy mode when I enter a specific zone. But when I enter the zone, the high accuracy mode stays off.
However, if I deselect the zone in the configuration menu of the companion app, then the high accuracy mode immediatly switch on. This happens whether the device is inside the zone or outside. (I did the same test in both cases)

Describe the behavior you expected
I expect the correct operation is that when the device enters the zone, the high accuracy mode is activated. And is deactivated when the device leaves the zone.

Then as an implementation, it could be useful to be able to activate/deactivate the high accuracy mode via a switch entity

Steps to reproduce the issue
1- After configuring a zone in the system, and configuring the complementary app to activate the high accuracy mode when the device enters the created zone

2- Enter the device into the area (high accuracy remains off)

3- Remove the device from the area (high accuracy remains off)

4- with the device inside the zone, deselect all the zones in the "background position" menu of the companion app (high accuracy mode is activated)

5- with the device inside the zone, select one or more zones in the "background position" menu of the companion app (high accuracy mode is deactivated)

6- repeat points 4 and 5 with the device outside the area, and the result is the same.

Android version: 13 TKQ1.221013.002

Device model: Xiaomi Redmi Note 10 Pro

Home Assistant version: 2024.10.3

**Comment by dshokouhi** [comment]
what happened to the issue template? this issue is now missing important information like logs and we cannot help without logs.

**Comment by dshokouhi** [comment]
your desscripion is also incredibly difficult to follow....i have no idea what this issue is describing.

**Comment by DOK87** [comment]
I apologize for my bad English, and the use of Google Translator. But your response is very very rude.
I only noticed a bug in the functioning of the app and I thought it was right to report it in everyone's interest. However, if this is a bother to you... no problem, close the thread.

My description is unclear... Maybe. As I said my English is not good, and I'm using google translate. Just tell me what isn't clear and I'll try to explain better.

Unfortunately, we are not all programmers or developers. And I'm not that much of an expert user. If you need the logs, no problem. Give me an indication of what logs you need and where to find them. I'll bring them back to you right away.

Thank you so much

**Comment by dshokouhi** [comment]
> I apologize for my bad English, and the use of Google Translator. But your response is very very rude.

if you are using google translator then you really dont know if my response was rude. I simply said the issue template was removed and I could not understand you as the text is difficult to follow. You should expect responses like that if you dont speak english and rely on a tool to translate and convey meaning.




> Just tell me what isn't clear and I'll try to explain better.

the whole text is hard to follow and uses terms not common with home assistant. Maybe a video recording will be easier to follow?



> If you need the logs, no problem. Give me an indication of what logs you need and where to find them. I'll bring them back to you right away.

the logs were mentioned in the issue template you had removed when creating the issue, the text there was meant to provide information and guide you in the data the issue needs. Logs are required  here.

Settings > companion app > troubleshooting > show and share logs

from there you need to replicate the issue and then grab the logs, they do not go far back either maybe only 10 minutes.

**Comment by DOK87** [comment]
thanks for the reply. And I apologize again for not understanding each other.

I will soon return home from the office and give you a more precise description using the new issue template.
Thank you

**Comment by DOK87** [comment]
> 
> 
> 
> **Home Assistant Android app version(s):**

2024.10.3-full

> **Android version(s):**

13 TKQ1.221013.002

> **Device model(s):**

Xiaomi Redmi Note 10 Pro - M2101K6G

> **Home Assistant version:**

2024.10.3

> **Last working Home Assistant release (if known):**
> 
> **Description of problem, include YAML if issue is related to notifications:**

I configured my home assistant with 2 concentric zones. Then I configured my app to activate the high accuracy mode in the background when the device enters the first zone, in order to verify exactly and immediately the entry into the second zone.

I noticed that configured like this, the high accuracy mode does not work as it should. When the device enters the first zone, the high accuracy mode sometimes does not activate. Other times it activates only for a few seconds, and then deactivates.

So I tried to do a test, deselecting all the zones in the "background position - High accuracy mode only when entering the zone" menu, the high accuracy mode activates.
When I select a zone, however, it immediately deactivates.
This happens both when the device is inside the zone and when it is outside.

> 
> **Companion App Logs:**

[homeassistant_companion_log_9-22-2024_18-13-58.txt](https://github.com/user-attachments/files/17479410/homeassistant_companion_log_9-22-2024_18-13-58.txt)

> **Screenshot or video of problem:**

![Immagine 2024-10-22 182227](https://github.com/user-attachments/assets/d05cab0f-7084-4bff-9a57-cf6c12442637)

> **Additional information:**

**Comment by dshokouhi** [comment]
one additional question are your zones overlapping? or are they in completely different locations?

**Comment by dshokouhi** [comment]
overlapping zones will indeed cause an issue and unexpected behavior as a result, do you see the same issue if your zones no longer overlap?

**Comment by DOK87** [comment]
I confirm. Without overlapping zones work.
I try this:

I moved the first zone (the larger one) away from the second zone (the smaller one). The device is currently inside the smaller zone.

After doing this, high accuracy was still off.

Then I tried, in the background position menu, to uncheck the zone I moved, leaving the check mark only on the zone I am in. By doing this, high accuracy was activated.

What I was trying to do is:
With high accuracy mode off, the tracker's position update is too slow. But keeping high accuracy mode on all the time is a waste of power.
So I wanted to use this feature to activate high accuracy mode near the entrance of the largest zone ("trigger to activate high accuracy"). With high accuracy mode on, I can then precisely and quickly control the entrance to the second zone. At that point I can deactivate high accuracy mode to save battery.

Of course, to do this, I absolutely need two overlapping zones. Especially since the zones are only circular.

I could work around this problem somehow if it were possible to activate/deactivate high accuracy via entity switch, but that is not possible.

**Comment by DOK87** [comment]
wow. I didn't know this, it can definitely be very useful. Also for other projects.

At this point I only have one question:
Is this issue with overlapping zones actually a bug? and will it be fixed in the future?

Thank you so much for the assistance

**Comment by dshokouhi** [comment]
> Is this issue with overlapping zones actually a bug? 

its a known issue actually https://github.com/home-assistant/android/issues/2966 

going to close this now as this seems to be identical to this.
